pkg/app/piped/cloudprovider/lambda/function.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lambda
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/yaml"
+)
+
+// FunctionManifest contains configuration for LambdaFunction.
+type FunctionManifest struct {
+	Name     string
+	ImageURI string
+}
+
+func loadFunctionManifest(path string) (FunctionManifest, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return FunctionManifest{}, err
+	}
+	return parseFunctionManifest(data)
+}
+
+func parseFunctionManifest(data []byte) (FunctionManifest, error) {
+	var obj unstructured.Unstructured
+	if err := yaml.Unmarshal(data, &obj); err != nil {
+		return FunctionManifest{}, err
+	}
+
+	imageURI, ok, err := unstructured.NestedString(obj.Object, "spec", "template", "spec", "image")
```

Comment:
I am not sure why we should still use this `unstructured` package to parse the manifest.
We can parse it directly from the YAML by updating the manifest struct to be like

https://github.com/kubernetes/api/blob/master/apps/v1/types.go#L254-L262

Reply:
In case of keep nested struct as the current sample `function.yaml` we need to define all sub struct to be able to unmarshal yaml content into `FunctionManifest` struct, I think it the cons 😅 
Should we simplify the current LambdaFunction manifest 👀 
```yaml
apiVersion: pipecd.dev/v1beta1
kind: LambdaFunction
metadata:
  name: SimpleFunction
spec:
  template:
    metadata:
      tags:
        app: simple
    spec:
      image: ecr.ap-northeast-1.amazonaws.com/lambda-test:v0.0.1
```

Reply:
Yes. I think we can simplify it.

``` yaml
apiVersion: pipecd.dev/v1beta1
kind: LambdaFunction
spec:
   name: SimpleFunction
   image: ecr.ap-northeast-1.amazonaws.com/lambda-test:v0.0.1
```

Reply:
got it, thank you 👍